### PR TITLE
Made Ray and its dependency Ray[Rllib] Optional. 

### PR DESCRIPTION
**Pull request by RutvikGupta**
Closes Issue #917.

### CODE REVIEW

**Comment by Adaickalavan** [comment]
For running multiple parallel SMARTS instances, I feel that instead of `examples/rayless_multi_instance.py` , we should provide a comprehensive easy-to-use environment wrapper along the lines of https://www.tensorflow.org/agents/api_docs/python/tf_agents/environments/ParallelPyEnvironment . This would allow us to better control features and handle issues in running multiple parallel SMARTS instances.

**Comment by RutvikGupta** [comment]
> For running multiple parallel SMARTS instances, I feel that instead of [`examples/rayless_multi_instance.py`](https://github.com/huawei-noah/SMARTS/blob/454b3df722c8d31a82313878a9eaee658c2a273e/examples/rayless_multi_instance.py) , we should provide a comprehensive easy-to-use environment wrapper along the lines of https://www.tensorflow.org/agents/api_docs/python/tf_agents/environments/ParallelPyEnvironment . This would allow us to better control features and handle issues in running multiple parallel SMARTS instances.

Could you please point me to an example of the following wrapper to know what exactly you meant by this?

**Comment by RutvikGupta** [comment]
> For running multiple parallel SMARTS instances, I feel that instead of [`examples/rayless_multi_instance.py`](https://github.com/huawei-noah/SMARTS/blob/454b3df722c8d31a82313878a9eaee658c2a273e/examples/rayless_multi_instance.py) , we should provide a comprehensive easy-to-use environment wrapper along the lines of (i) [ParallelPyEnvironment](https://www.tensorflow.org/agents/api_docs/python/tf_agents/environments/ParallelPyEnvironment) and (ii) [ProcessPyEnvironment](https://www.tensorflow.org/agents/api_docs/python/tf_agents/environments/parallel_py_environment/ProcessPyEnvironment) . This would allow us to better control features and handle issues in running multiple parallel SMARTS instances.

Opened Issue #964 for this.

**Comment by Adaickalavan** [comment]
Considering the creation of issue #964 , I would recommend removing (i) `examples/rayless_multi_instance.py` and (ii) `def test_rayless_multi_instance_example()` in `tests/test_examples.py`, from this pull request.

File: examples/__init__.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class RayException(Exception):
+    """An exception raised if ray package is required but not available."""
+
+    @classmethod
+    def required_to(cls, thing):
+        return cls(
+            f"""Ray Package is required to simulatre {thing}.
```
Comment (Adaickalavan) [inline examples/__init__.py:7]:
```suggestion
            f"""Ray Package is required to simulate {thing}.
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -16,6 +16,7 @@ Copy and pasting the git commit messages is __NOT__ enough.
 - Added a new utility experiment file `cli/run.py` to replace the context given by `supervisord.conf`. See PR #911.
 - Added `scl zoo install` command to install zoo policy agents at the specified paths. See Issue #603.
 - Added a `FrameStack` wrapper which returns stacked observations for each agent.
+- Added a `rayless_multi_instance.py` example that does not use ray to train in remote. See Issue #917.
```
Comment (Adaickalavan) [inline CHANGELOG.md:19]: This example has been removed.

File: examples/__init__.py [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+               You may not have installed the [train] or [test] dependencies required to run the ray dependent example.
+               Install them first using the command `pip install -e .[train, test]` at the source directory. to install the package ray[rllib]==1.0.1.post1"""
```
Comment (Adaickalavan) [inline examples/__init__.py:9]:
```suggestion
               Install them first using the command `pip install -e .[train, test]` at the source directory, to install the package ray[rllib]==1.0.1.post1"""
```

**Review by Adaickalavan — APPROVED** [review]

**Review by sah-huawei — APPROVED** [review]

**Pull request merged**